tests/bud.bats

```diff
@@ -117,6 +117,21 @@ symlink(subdir)"
   expect_output --substring $(realpath "$BUDFILES/dockerignore3/.dockerignore")
 }
 
+@test "build with basename resolving default arg" {
+  run_buildah info --format '{{.host.arch}}'
+  myarch="$output"
+
+  run_buildah build --platform linux/$myarch $WITH_POLICY_JSON -t test -f $BUDFILES/base-with-arg/Containerfile
```

Comment:
Pulling out the variant and adding a ${variant:+/${variant}} to the end of the --platform value should help the arm/v6 and arm/v7 cases.